GitHubProjects.ps1

```diff
@@ -0,0 +1,751 @@
+
+        [Parameter(Mandatory, ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(Mandatory, ParameterSetName = 'Uri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory, ParameterSetName = 'Organization')]
+        [string] $OrganizationName,
+
+        [Parameter(Mandatory, ParameterSetName = 'User')]
+        [string] $UserName,
+
+        [Parameter()]
+        [string] $Name,
+
+        [Parameter(Mandatory, ParameterSetName = 'Project')]
+        [int64] $Project,
+
+        [Parameter(ParameterSetName = 'Elements')]
+        [Parameter(ParameterSetName = 'Organization')]
+        [Parameter(ParameterSetName = 'User')]
+        [Parameter(ParameterSetName = 'Uri')]
+        [ValidateSet('Open', 'Closed', 'All')]
+        [string] $State,
+
+        [Parameter()]
+        [string] $AccessToken,
+
+        [Parameter()]
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{}
+
+    $uriFragment = [String]::Empty
+    $description = [String]::Empty
+    if ($PSCmdlet.ParameterSetName -eq 'Project')
+    {
+
+        $telemetryProperties['Project'] = $Project
+
+        $uriFragment = "/projects/$Project"
+        $description = "Getting project $project"
+    }
+    else
+    {
+        if ($PSCmdlet.ParameterSetName -in ('Elements', 'Uri'))
+        {
+            $elements = Resolve-RepositoryElements
+            $OwnerName = $elements.ownerName
+            $RepositoryName = $elements.repositoryName
+
+            $telemetryProperties['OwnerName'] = Get-PiiSafeString -PlainText $OwnerName
+            $telemetryProperties['RepositoryName'] = Get-PiiSafeString -PlainText $RepositoryName
+
+            $uriFragment = "/repos/$OwnerName/$RepositoryName/projects"
+            $description = "Getting projects for $RepositoryName"
+        }
+        elseif ($PSCmdlet.ParameterSetName -eq 'Organization')
+        {
+
+            $telemetryProperties['OrganizationName'] = Get-PiiSafeString -PlainText $OrganizationName
+
+            $uriFragment = "/orgs/$OrganizationName/projects"
+            $description = "Getting projects for $OrganizationName"
+        }
+        elseif ($PSCmdlet.ParameterSetName -eq 'User')
+        {
+
+            $telemetryProperties['UserName'] = Get-PiiSafeString -PlainText $UserName
+
+            $uriFragment = "/users/$UserName/projects"
+            $description = "Getting projects for $UserName"
+        }
+
+        if ($PSBoundParameters.ContainsKey('Project'))
+        {
+            $uriFragment = "$uriFragment/$Project"
+            $description = "Getting project $Project"
+        }
+        elseif ($PSBoundParameters.ContainsKey('State'))
+        {
+            $getParams = @()
+            $State = $State.ToLower()
+            $getParams += "state=$State"
+
+            $uriFragment = "$uriFragment`?" + ($getParams -join '&')
+            $description = "Getting projects for $RepositoryName"
```

Comment:
Good catch
changed to add on to description since there will always have to be something already in description
```
$description += " with state '$state'"
```

![image](https://user-images.githubusercontent.com/981370/80517198-9bbc9980-897c-11ea-94a9-800ee1d5db68.png)